### PR TITLE
777 cohort workflow subjects

### PR DESCRIPTION
**Pull request by annacanal**
- [x] folder src/cohort + changes to braph2, genesis, test, etc...
- [x] _Group.gen.m, _Subject.gen.m (see attached files, modify as needed)
[cohort.zip](https://github.com/giovannivolpe/Braph-2.0-Genesis/files/5910842/cohort.zip)
- [x] folder worflows/connectivity, functional, structural + changes as needed

- [x] _SubjectCON.gen.m + import + export + example files
- [x] CON import/export TXT

- [x] _SubjectFUN.gen.m + import + export + example files
- [x] FUN import/export TXT

- [x] _SubjectST.gen.m + import + export + example files
- [x] ST import/export TXT

### CODE REVIEW

File: braph2genesis/braph2/braph2.m [hunk 1]
```diff
@@ -32,13 +32,13 @@
...
-% addpath(workflows_dir)
-% workflows_dir_list = dir(workflows_dir);   % get the folder contents
-% workflows_dir_list = workflows_dir_list([workflows_dir_list(:).isdir] == 1);  % remove all files (isdir property is 0)
-% workflows_dir_list = workflows_dir_list(~ismember({workflows_dir_list(:).name}, {'.', '..'}));  % remove '.' and '..'
-% for i = 1:1:length(workflows_dir_list)
-%     addpath([workflows_dir filesep workflows_dir_list(i).name])
-% end
+workflows_dir = [fileparts(which('braph2')) filesep 'workflows'];
+addpath(workflows_dir)
+workflows_dir_list = dir(workflows_dir);   % get the folder contents
```
Comment (giovannivolpe) [inline braph2genesis/braph2/braph2.m:37]: leave only one space before % (new approach)

File: braph2genesis/src/cohort/_Group.gen.m [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+SUB_CLASS (parameter, class) is the class of the subjects of the group.
+%%%% ¡settings!
+'Subject'
+
+%%% ¡prop!
+SUB_DICT (data, idict) is an indexed dictionary containing the subjects of the group.
+%%%% ¡settings!
+'Subject'
+%%%% ¡default!
+IndexedDictionary('IT_CLASS', 'Subject', 'IT_KEY', 1)
```
Comment (giovannivolpe) [inline braph2genesis/src/cohort/_Group.gen.m:35]: This should not be needed as it's the default value

File: braph2genesis/src/cohort/_Group.gen.m [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    %
+    % REMOVESUBJECTS(GR, SUBJECTS) removes the subjects from the group.
+    %
+    % See also subjectnumber, contains_subject, addSubject.
+    
+    for i = 1:1:length(subjects)
+        subject = subjects{i};
+        gr.removeSubject(subject)
+    end
+end
```
Comment (giovannivolpe) [inline braph2genesis/src/cohort/_Group.gen.m:105]: remove because they can be called from SUB_DICT

File: braph2genesis/src/cohort/_Subject.gen.m [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+function subject_list = getList()
+    % GETLIST returns the list of available subjects
+    %
+    % SUBJECT_LIST = GETLIST() returns the list of subjects
+    % (cell array) that are subclasses of Subject.
+    %
+    % See also getClass, getName, getDescription.
+
+    subject_list = subclasses('Subject');
+end
```
Comment (giovannivolpe) [inline braph2genesis/src/cohort/_Subject.gen.m:33]: Not needed.
One can use directly subclasses('Subject')

File: braph2genesis/test/test_braph2.m [hunk 5]
```diff
@@ -19,15 +19,15 @@
...
-% workflows_dir_list = workflows_dir_list(~ismember({workflows_dir_list(:).name}, {'.', '..'}));  % remove '.' and '..'
-% for i = 1:1:length(workflows_dir_list)
-%     directories_to_test{end + 1} = [workflows_dir filesep workflows_dir_list(i).name]; %#ok<SAGROW>
-% end
+workflows_dir = [fileparts(which('braph2')) filesep 'workflows'];
+
+addpath(workflows_dir)
+workflows_dir_list = dir(workflows_dir);   % get the folder contents
+workflows_dir_list = workflows_dir_list([workflows_dir_list(:).isdir] == 1);  % remove all files (isdir property is 0)
+workflows_dir_list = workflows_dir_list(~ismember({workflows_dir_list(:).name}, {'.', '..'}));  % remove '.' and '..'
```
Comment (giovannivolpe) [inline braph2genesis/test/test_braph2.m:27]: single space before %

**Review by giovannivolpe — COMMENTED** [review]
add gitignore for .DS_store files

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
Please, make sure you also add .DS_Store files to the gitignore file

File: braph2genesis/src/cohort/_Group.gen.m (hunk 3 above)
Comment (annacanal) [reply braph2genesis/src/cohort/_Group.gen.m:105]: ok

**Review by c-yuwei — APPROVED** [review]

**Review by giovannivolpe — COMMENTED** [review]
Let's check it together and then merge it

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**